src/org/centos/pipeline/PipelineUtils.groovy

```diff
@@ -160,4 +169,254 @@ def setMessageFields(messageType){
+    // TODO: Determine if this method can be removed. No usages seen in code.
+    sendCIMessage messageContent: msgContent,
+            messageProperties: msgProps,
+
+            messageType: 'Custom',
+            overrides: [topic: "${topic}"],
+            providerName: "${MSG_PROVIDER}"
+}
+
+/**
+ * Library to parse CI_MESSAGE and inject its key/value pairs as env variables.
+ *
+ */
+def injectFedmsgVars() {
+
+    // Parse the CI_MESSAGE into a Map
+    def ci_data = new JsonSlurper().parseText(env.CI_MESSAGE)
+
+    // If we have a 'commit' key in the CI_MESSAGE, for each key under 'commit', we
+    // * prepend the key name with fed_
+    // * replace any '-' with '_'
+    // * truncate the value for the key at the first '\n' character
+    // * replace any double-quote characters with single-quote characters in the value for the key.
+
+    if (ci_data['commit']) {
+        ci_data.commit.each { key, value ->
+            env."fed_${key.toString().replaceAll('-', '_')}" =
+                    value.toString().split('\n')[0].replaceAll('"', '\'')
+        }
+        if (env.fed_branch == 'master'){
+            env.branch = 'rawhide'
+        } else {
+            env.branch = env.fedbranch
+        }
+    }
+}
+
+/**
+ * Library to set default environmental variables. Performed once at start of Jenkinsfile
+ */
+def setDefaultEnvVars(){
+    env.MAIN_TOPIC = env.MAIN_TOPIC ?: 'org.centos.prod'
+    env.MSG_PROVIDER = env.MSG_PROVIDER ?: 'fedora-fedmsg'
+    env.HTTP_BASE = env.HTTP_BASE ?: 'http://artifacts.ci.centos.org/artifacts/fedora-atomic'
+    env.RSYNC_USER = env.RSYNC_USER ?: 'fedora-atomic'
+    env.RSYNC_SERVER = env.RSYNC_SERVER ?: 'artifacts.ci.centos.org'
+    env.RSYNC_DIR = env.RSYNC_DIR ?: 'fedora-atomic'
+    env.basearch = env.basearch ?: 'x86_64'
+    env.OSTREE_BRANCH = env.OSTREE_BRANCH ?: ''
+    env.commit = env.commit ?: ''
+    env.image2boot = env.image2boot ?: ''
+    env.image_name = env.image_name ?: ''
+    env.FEDORA_PRINCIPAL = env.FEDORA_PRINCIPAL ?: 'bpeck/jenkins-continuous-infra.apps.ci.centos.org@FEDORAPROJECT.ORG'
+    env.package_url = env.package_url ?: ''
+    env.nvr = env.nvr ?: ''
+    env.original_spec_nvr = env.original_spec_nvr ?: ''
+    env.ANSIBLE_HOST_KEY_CHECKING = env.ANSIBLE_HOST_KEY_CHECKING ?: 'False'
+}
+
+/**
+ * Library to set stage specific environmental variables
+ *
+ * variables
+ *  currentStage - current stage running
+ */
+def setStageEnvVars(currentStage){
+    // branch, basearch, fed_repo, & fed_rev should be accessible after calling injectFedmsgVars()
+
+    def stages =
+            ["ci-pipeline-rpmbuild"                : [
+                    task                     : "./ci-pipeline/tasks/rpmbuild-test",
+                    playbook                 : "ci-pipeline/playbooks/setup-rpmbuild-system.yml",
+                    ref                      : "fedora/${branch}/${basearch}/atomic-host",
+                    repo                     : "${fed_repo}",
```

Comment:
Does the vars here need to be env.<varname>?

Reply:
So I posed this question to the team but didn't really hear a consensus, so I'll ask for your feedback here:

When you set env.VAR_NAME = "value", you're modifying the global env var. As such, calling ${VAR_NAME} would be resolved as env.VAR_NAME *unless you'd defined VAR_NAME in a more local scope*. If you'd not set VAR_NAME in your script or method scope, called env.VAR_NAME and VAR_NAME results in referencing the same value.

Personally, I think it makes sense to explicitly reference the env.VAR_NAME.

Reply:
I agree if we are setting them in the injection then we should use env.*